### PR TITLE
fix: phone navbar not properly handling closing state with country-selector

### PR DESCRIPTION
**Pull request by Primdahl26**
This fixes selecting country on mobile devices - before the burger dropdown menu would be locked in a state where it thought it was closed, but was actually open. This fix was to change the country selector to also handle the state of the dropdown.

Also adds some cool new transitions between the `x` and the burger icon. Also added this to filters and theme selector.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: frontend/src/lib/components/country_selector.svelte [hunk 1]
```diff
@@ -5,6 +5,7 @@
   import { get } from "svelte/store"
 
   export let countries: main_Country[]
+  export let handleDropdownClick: Function | undefined = undefined
```
Comment (AndreasPB) [inline frontend/src/lib/components/country_selector.svelte:8]: This seems redundant - Does it have to default to `undefined`?

```suggestion
  export let handleDropdownClick: Function
```

**Review by AndreasPB — APPROVED** [review]

File: frontend/src/lib/stores/preferences.ts [hunk 2]
```diff
@@ -14,6 +14,10 @@ currentGenres.subscribe(value => {
   }
 })
 
+export const isBurgerMenuOpen = writable<boolean>(
+  (browser && localStorage.isBurgerMenuOpen) || false
+)
+
```
Comment (AndreasPB) [inline frontend/src/lib/stores/preferences.ts:20]: We don't need a special store for this, just use the normal one

Comment (Primdahl26) [reply frontend/src/lib/stores/preferences.ts]: Done

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**